scanpipe/tests/regen_test_data.py

```diff
@@ -149,3 +151,32 @@ def test_regen_asgiref_test_files(self):
                 "package": True,
             },
         )
+
+    def test_regenerate_scorecard_data(self):
+        """
+        Regenerate and save scorecard data by directly calling the OSSF Scorecard
+        API
+        """
+        scorecard_data_file = self.data / "scorecode" / "scorecard_response.json"
+        platform, org, repo = "github.com", "nexB", "scancode-toolkit"
+
+        OSSF_SCORECARD_API_URL = "https://api.securityscorecards.dev"
+
+        url = f"{OSSF_SCORECARD_API_URL}/projects/{platform}/{org}/{repo}"
+
+        try:
+            # Fetch the scorecard data from the API
+            response = requests.get(url, timeout=10)
+            response.raise_for_status()
+            scorecard_data = response.json()
```

Comment:
You should use `scorecode.ossf_scorecard.fetch_scorecard` API to avoid code duplication?